pyexchange/kraken.py

```diff
@@ -0,0 +1,301 @@
+                     amount=Wad.from_number(trade[1]))
+
+
+class KrakenApi(PyexAPI):
+    """Kraken API interface.
+    https://www.kraken.com/features/api
+    """
+
+    logger = logging.getLogger()
+
+    def __init__(self, api_server: str, api_key: str, secret_key: str, timeout: float):
+        assert(isinstance(api_key, str))
+        assert(isinstance(secret_key, str))
+
+        self.api_server = api_server
+        self.api_key = api_key
+        self.secret_key = secret_key
+        self.timeout = timeout
+
+    def get_markets(self):
+        return self._http_unauthenticated("POST", "/0/public/AssetPairs", {})
+
+    def get_assets(self):
+        return self._http_unauthenticated("POST", "/0/public/Assets", {})
+
+    def get_pair(self, pair):
+        assert(isinstance(pair, str))
+        return self.get_markets().get(pair)
+
+    def get_balances(self):
+        return self._http_authenticated("POST", "/0/private/Balance", {})
+
+    def get_trade_balances(self):
+        return self._http_authenticated("POST", "/0/private/TradeBalance", {})
+
+    def get_orders(self, pair: str) -> List[Order]:
+        assert(isinstance(pair, str))
+
+        result = self._http_authenticated("POST", "/0/private/OpenOrders", {})
+        orders = []
+
+        for order_id, order in result['open'].items():
+            if order['descr']['pair'] == pair and order['descr']['ordertype'] == 'limit':
+                orders.append(Order.from_our_order(order_id, order))
+
+        return orders
+
+    def place_order(self, pair: str, is_sell: bool, price: Wad, amount: Wad) -> str:
+        assert(isinstance(pair, str))
+        assert(isinstance(is_sell, bool))
+        assert(isinstance(price, Wad))
+        assert(isinstance(amount, Wad))
+
+        data = {
+            'pair': pair,
+            'type': 'sell' if is_sell else "buy",
+            'ordertype': 'limit',
+            "volume": str(amount),
+            "price": str(price)
+        }
+
+        self.logger.info(f"Placing order ({data['type']}, amount {data['volume']} of {pair},"
+                         f" price {data['price']})...")
+
+        result = self._http_authenticated("POST", "/0/private/AddOrder", data)
+
+        return result['txid'][0]
+
+    def cancel_order(self, order_id: str) -> bool:
+        assert(isinstance(order_id, str))
+
+        self.logger.info(f"Cancelling order #{order_id}...")
+        self._http_authenticated("POST", "/0/private/CancelOrder", {'txid': order_id})
+
+        return True
+
+    def get_trades(self, pair: str, page_number: int = 1) -> List[Trade]:
+        assert(isinstance(pair, str))
+        assert(isinstance(page_number, int))
+        assert(page_number == 1)
+
+        result = self._http_authenticated("POST", f"/0/private/TradesHistory", {})
+
+        trades = []
+
+        for trade_id, trade in result['trades'].items():
+            if trade['pair'] == pair and trade['ordertype'] == 'limit':
```

Comment:
No need to specify `trade['ordertype'] == 'limit':` all orders should already be limit orders no?

Reply:
there could be multiple types of orders

market
    limit (price = limit price)
    stop-loss (price = stop loss price)
    take-profit (price = take profit price)
    stop-loss-profit (price = stop loss price, price2 = take profit price)
    stop-loss-profit-limit (price = stop loss price, price2 = take profit price)
    stop-loss-limit (price = stop loss trigger price, price2 = triggered limit price)
    take-profit-limit (price = take profit trigger price, price2 = triggered limit price)
    trailing-stop (price = trailing stop offset)
    trailing-stop-limit (price = trailing stop offset, price2 = triggered limit offset)
    stop-loss-and-limit (price = stop loss price, price2 = limit price)
    settle-position

Reply:
yes but what will we be placing... Only `limit` orders right

Reply:
Correct, so we just want to fetch limit orders